draft-ietf-quic-transport.md

```diff
@@ -2432,17 +2432,18 @@ The capacity available on the new path might not be the same as the old path.
 Packets sent on the old path MUST NOT contribute to congestion control or RTT
 estimation for the new path.
 
-On confirming a peer's ownership of its new address, an endpoint MUST
+On confirming a peer's ownership of its new address, an endpoint SHOULD
 immediately reset the congestion controller and round-trip time estimator for
 the new path to initial values (see Appendices A.3 and B.3 in {{QUIC-RECOVERY}})
-unless it has knowledge that a previous send rate or round-trip time estimate is
-valid for the new path.  For instance, an endpoint might infer that a change in
-only the client's port number is indicative of a NAT rebinding, meaning that the
-new path is likely to have similar bandwidth and round-trip time. However, this
-determination will be imperfect.  If the determination is incorrect, the
-congestion controller and the RTT estimator are expected to adapt to the new
-path.  Generally, implementations are advised to be cautious when using previous
-values on a new path.
+unless the only change is the other endpoint's port number.  Because
```

Comment:
On the mailing list, I asked whether I should add "and the connection ID did not change" to the end of this sentence. Opinions?

Reply:
I think not.  Implementations that know they're likely to have a NAT rebinding (i.e. after quiescence) are advised to change the CID as well.  That doesn't make it less a NAT rebinding.

Reply:
The point is that _if_ the CID changed, it was an active migration attempt, which indicates that the peer at least attempted to switch paths.

Reply:
Yeah, I agree that it's consistent with the idea that the peer likely changed its attachment point. I don't think this makes a difference in practice, but it seems consistent with our principles here.

It does make it a bit more work to check, but it's not that hard.

Reply:
Correctly predicting a NAT rebind isn't an "attempt" to switch paths.  It's using the same path, but making a guess about what might have happened.

If you're not going to reset the congestion controller when the port changes without the peer realizing it, and you're not going to reset the congestion controller if the peer guesses they're about to get rebound but doesn't, then I don't think it makes sense to reset it if they guess and are right.

Reply:
That's fair, but why would a peer change the CID for a NAT rebinding?

Reply:
We encourage changing the CID and port to "force" a NAT rebinding if you've been idle (Section 9.5) in order to reduce linkability.  In the same breath, we note that this might reset congestion controller state, so be careful about changing port too often.  A conservative implementation of that guidance might change CID but not port, letting NAT rebindings happen if they happen.

Reply:
I don't see how the text in Section 9.5 conflicts with the proposed change?

Reply:
Whether an endpoint passively experiences or deliberately forces a NAT rebind is indistinguishable at the server side.  Changing CID tells you nothing more than that they wanted to reduce their linkability if they got rebound.  A change in CID doesn't imply a path change; a change in endpoint address is the only reliable guide on that.

We're saying that you SHOULD reset unless there's a change that's broader than the port, because a port-only change is very likely to be the same path.  A CID change is even less likely to indicate a path change.  Therefore, it should not be a cause for resetting the congestion controller.

Reply:
@MikeBishop: I see your point. The concern @larseggert was trying to address was that in some carrier-grade NATs, it is possible that what appears as a port change might actually be a path change on the private side of the CGNAT, but your point is that CID change cannot be assumed to be a good proxy for path change. At a minimum, we don't have experience with this yet, so we can't assume it.

I don't have a strong opinion one way or the other here. In practice, I completely expect people to treat port changes as same path. Given @MikeBishop's point about CID changes also not reflecting a path change, I would suggest leaving the PR as it is.